File: python/compare_version_numbers.py
class Solution:
    # @param version1, a string
    # @param version2, a string
    # @return an integer
    def compareVersion(self, version1, version2):
        if not version1 or not version2:
            raise ValueError('wtf')

        v1, v2 = version1.split('.'), version2.split('.')
        common = max(len(v1), len(v2))

        for i in xrange(common):
            token1 = int(v1[i]) if i < len(v1) else 0
            token2 = int(v2[i]) if i < len(v2) else 0
            if token1 < token2:
                return -1
            elif token1 > token2:
                return 1
                
        return 0

File: java/CompareVersionNumbers.java
public class CompareVersionNumbers {
    public int compareVersion(String version1, String version2) {
        if (version1 == null || version2 == null) {
            throw new IllegalArgumentException();
        }

        // bug 1: split by dot must use regx
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");

        int i = 0, j = 0;
        while (i < v1.length || j < v2.length) {
            int a = i < v1.length ? Integer.parseInt(v1[i]) : 0;
            int b = j < v2.length ? Integer.parseInt(v2[j]) : 0;
            if (a > b) {
                return 1;
            } else if (a < b) {
                return -1;
            }
        }

        return 0;
    }
}